examples/diff.c

```diff
@@ -335,3 +357,105 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+void die(char* message){
+	fprintf(stderr, message);
+	exit(EXIT_FAILURE);
+}
+
+struct pair extract_from_string(const char *cmd_opt)
+{
+	struct pair p;
+	char **endptr;
+	const char * cmd_opt2;
+	int tmp = strtol(cmd_opt,endptr,'A');
+
+
+	if (tmp==0) {
+		if (*endptr==cmd_opt) {
+			p.first = DEFAULT_THRESHOLD;
+			p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+		} else if (**endptr=='/') {
+			p.first = DEFAULT_THRESHOLD;
+			cmd_opt2 = *endptr++;
+			tmp = strtol(cmd_opt2,endptr,'A');
+			if (tmp==0) {
+				if (*endptr == cmd_opt2) {
+					p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+				} else if (**endptr != '\0') {
+					die("Invalid input format");
+				} else {
+					p.second = tmp;
+				}
+			} else {
+				p.second = tmp;
+			}
+		} else if (**endptr != '\0') {
+			die("Invalid input format");
+		}
+	} else if (**endptr=='%') {
+		p.first = tmp;
+		endptr++;
+		if (**endptr=='/') {
+			cmd_opt2 = *endptr++;
+			tmp = strtol(cmd_opt2,endptr,'A');
+			if (tmp==0) {
+				if (*endptr == cmd_opt2) {
+					p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+				} else if (**endptr != '\0') {
+					die("Invalid input format");
+				} else {
+					p.second = tmp;
+				}
+			} else {
+				p.second = tmp;
+			}
+		} else if (**endptr == '\0') {
+			p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+		}
+	} else if (**endptr=='/') {
+		p.first = tmp;
+		cmd_opt2 = *endptr++;
+		tmp = strtol(cmd_opt2,endptr,'A');
+		if (tmp==0) {
+			if (*endptr == cmd_opt2) {
+				p.second = DEFAULT_BREAK_REWRITE_THRESHOLD;
+			} else if (**endptr != '\0') {
+				die("Invalid input format");
+			} else {
+				p.second = tmp;
+			}
+		} else {
+			p.second = tmp;
+		}
+	} else if (**endptr!='\0') {
+		die("Invalid input format");
+	}
+
+	return p;
+}
+
+struct pair extract_m_n(struct args_info *args, const char *opt)
```

Comment:
This still returns a struct ? I think `int extract_break_rewrites(int *m, int *n, const char *arg)` would be nicer. Also, if using `sscanf` above makes it shorter, it could be merged in this one instead of being in a separate function.